FWCore/Services/plugins/SimpleMemoryCheck.cc

```diff
@@ -350,6 +355,22 @@ namespace edm {
 
       openFiles();
 
+      if (sampleEveryNSeconds_ > 0) {
+        if (oncePerEventMode_) {
+          throw cms::Exception("BadConfigOptions")
+              << "'sampleEventNSeconds' and 'oncePerEventMode' cannot be used together";
+        }
+        if (moduleSummaryRequested_) {
+          throw cms::Exception("BadConfigOptions")
+              << "'sampleEventNSeconds' and 'moduleSummaryRequested' cannot be used together";
+        }
+        iReg.watchPostBeginJob(this, &SimpleMemoryCheck::startSamplingThread);
+        iReg.watchPreEndJob(this, &SimpleMemoryCheck::stopSamplingThread);
```

Comment:
Does this need to listen the early termination signals too, or do we call `PreEndJob` after those?

Reply:
I guess with https://github.com/cms-sw/cmssw/pull/45434 we try to call EndJob if it is possible in any way (alternative being the job terminating via `std::terminate()` or similar)?

Reply:
I agree that the EndJob transition should be run after early termination.

Reply:
I agree also. That is what is should be doing.